coalib/processes/Processing.py

```diff
@@ -344,7 +347,35 @@ def instantiate_processes(section,
         log_printer,
         ignored_file_paths=glob_list(section.get('ignore', "")),
         limit_file_paths=glob_list(section.get('limit_files', "")))
+
+    # This stores all matched files irrespective of whether coala is run
+    # only on changed files or not. Global bears require all the files
+    complete_filename_list = filename_list
+
+    # Start tracking all the files
+    if cache:
+        cache.track_files(set(complete_filename_list))
+
+    # We need to compute the changed files so that the cache gets updated
+    # even if caching is disabled (in which case all files will be processed
+    # but the cache will be updated).
+    changed_files = cache.get_uncached_files(
+        set(filename_list)) if cache else filename_list
```

Comment:
don't compute this if caching is disabled please, performance is the goal here, right?